superset-frontend/src/views/CRUD/csstemplates/CssTemplateModal.tsx

```diff
@@ -0,0 +1,258 @@
+    const { target } = event;
+
+    const data = {
+      ...currentCssTemplate,
+      template_name: currentCssTemplate ? currentCssTemplate.template_name : '',
+      css: currentCssTemplate ? currentCssTemplate.css : '',
+    };
+
+    data[target.name] = target.value;
+    setCurrentCssTemplate(data);
+  };
+
+  const onCssChange = (css: string) => {
+    const data = {
+      ...currentCssTemplate,
+      template_name: currentCssTemplate ? currentCssTemplate.template_name : '',
+      css,
+    };
+    setCurrentCssTemplate(data);
+  };
+
+  const validate = () => {
+    if (
+      currentCssTemplate &&
+      currentCssTemplate.template_name.length &&
+      currentCssTemplate.css &&
+      currentCssTemplate.css.length
+    ) {
+      setDisableSave(false);
+    } else {
+      setDisableSave(true);
+    }
+  };
+
+  // Initialize
+  if (
+    isEditMode &&
+    (!currentCssTemplate ||
+      !currentCssTemplate.id ||
+      (cssTemplate && cssTemplate.id !== currentCssTemplate.id) ||
+      (isHidden && show))
+  ) {
+    if (cssTemplate && cssTemplate.id !== null && !loading) {
+      const id = cssTemplate.id || 0;
+
+      fetchResource(id).then(() => {
+        setCurrentCssTemplate(resource);
+      });
+    }
+  } else if (
+    !isEditMode &&
+    (!currentCssTemplate || currentCssTemplate.id || (isHidden && show))
+  ) {
+    setCurrentCssTemplate({
+      template_name: '',
+      css: '',
+    });
+  }
+
+  // Validation
+  useEffect(() => {
+    validate();
+  }, [
+    currentCssTemplate ? currentCssTemplate.template_name : '',
+    currentCssTemplate ? currentCssTemplate.css : '',
+  ]);
+
+  // Show/hide
+  if (isHidden && show) {
+    setIsHidden(false);
+  }
+
+  return (
+    <Modal
+      disablePrimaryButton={disableSave}
+      onHandledPrimaryAction={onSave}
+      onHide={hide}
+      primaryButtonName={isEditMode ? t('Save') : t('Add')}
+      show={show}
+      width="800px"
+      title={
+        <h4 data-test="css-template-modal-title">
+          {isEditMode ? (
+            <StyledIcon name="edit-alt" />
+          ) : (
+            <StyledIcon name="plus-large" />
+          )}
+          {isEditMode
+            ? t('Edit CSS Template Properties')
+            : t('Add CSS Template')}
+        </h4>
+      }
+    >
+      <StyledCssTemplateTitle>
+        <h4>{t('Basic Information')}</h4>
+      </StyledCssTemplateTitle>
+      <TemplateContainer>
+        <div className="control-label">
+          {t('css template name')}
+          <span className="required">*</span>
+        </div>
+        <input
+          name="template_name"
+          onChange={onTemplateNameChange}
+          type="text"
+          value={currentCssTemplate?.template_name}
+        />
+      </TemplateContainer>
+      <TemplateContainer>
+        <div className="control-label">
+          {t('css')}
+          <span className="required">*</span>
+        </div>
+        <StyledCssEditor
+          height="240px"
```

Comment:
Might not be worth the battle, but I'm curious if we could make the modal take up as much room on the screen as it can, and use a flexbox layout within the modal, which would let THIS component take up as much of the layout as possible, rather than limiting its height with a static value.